### PR TITLE
Migrate media

### PR DESCRIPTION
**Pull request by megoth**
Tried migrating media module to TypeScript.

Builds upon the work in https://github.com/solid/solid-ui/pull/170

### CODE REVIEW

**Comment by michielbdejong** [comment]
@megoth what is the status of this PR now?

**Comment by megoth** [comment]
Same as with https://github.com/solid/solid-ui/pull/169#issuecomment-717177811: Seems to be a simple conversion to TypeScript. But I see that there are conflicing files now, suggesting that the files I originally converted has changed later. I would close this PR because of that, and suggest doing another round of conversion if the team deem it necessary.

**Comment by angelo-v** [comment]
I resolved the merge conflicts and tested it locally. Someone should at least take a look at my [merge commit](https://github.com/solid/solid-ui/pull/171/commits/17b895e11697439c3f653ff827e01903ae7db00f) and perhaps someone else can do some additional testing. @michielbdejong @timbl @SharonStrats ?

File: src/create/create.ts [hunk 1]
```diff
@@ -44,7 +53,23 @@ export function newThingUI (
...
 
-  const selectNewTool = function (_event?) {
+  function styleTheIcons (style) {
+    for (let i = 0; i < iconArray.length; i++) {
+      let st = iconStyle + style
+      if (iconArray[i].disabled) {
+        // @@ unused
+        st += 'opacity: 0.3;'
+      }
+      iconArray[i].setAttribute('style', st) // eg 'background-color: #ccc;'
```
Comment (TallTed) [inline src/create/create.ts]:
```suggestion
      iconArray[i].setAttribute('style', st) // e.g., 'background-color: #ccc;'
```

**Review by timbl — APPROVED** [review]
Look good to me

**Pull request merged**